### DISCUSSION TITLE
A question about recursive directory searching

### DISCUSSION THREAD

**Discussion by s-p-turner** [BurntSushi/ripgrep#2367] [Q&A] [answered]
I'm currently using ripgrep 13.0.0.

I've noticed that the following commands don't actually search all dirs/files recursively and instead only print out a very small subset of matches:

```
rg 'string'
rg 'string' .
```

So far the only way I can instruct ripgrep to properly search recursively is to use:

`rg 'string' *`

Is this behaviour by design or could it be OS- or shell- related? (Ubuntu 20, bash shell.)

**Comment by BurntSushi** [comment, marked as answer]
Do the beginning sentences of the README answer your question? Emphasis mine:

> ripgrep is a line-oriented search tool that recursively searches the current directory for a regex pattern. **By default, ripgrep will respect gitignore rules and automatically skip hidden files/directories and binary files. (To disable all automatic filtering by default, use rg -uuu.)**

A similar message is repeated in the GitHub description of the project and the man page.

Using `*` circumvents some of the filtering because the shell expands `*` before invoking `rg`. And regardless of what filters are enabled, ripgrep always searches whatever is explicitly given to it.

**Comment by s-p-turner** [comment]
I've been working with a build system designed by a third party and I've discovered that it creates a .gitignore file in the root of the workspace. And this file is indeed the reason why ripgrep was only reporting a subset of matches.

Thank you for the swift reply. and apologies for wasting your time.

**Reply by BurntSushi** [reply]
No problem. See also the `--debug` flag. It will tell you why ripgrep skips any particular file (or directory).
